files/check_zfs

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+########################################################################
+##
+## Author: Michel Weitbrecht (michel.weitbrecht@stuvus.uni-stuttgart.de)
+## Adapted from https://github.com/zlacelle/nagios_check_zfs_linux
+## Licensed under GPL (see below)
+##
+## This program is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with this program.  If not, see <http://www.gnu.org/licenses/>.
+##
+########################################################################
+
+import subprocess
+import argparse
+from common import doCheck
+
+ZPOOL_COMMAND = '/sbin/zpool'
+POSSIBLE_STATES = ('OK', 'WARNING', 'CRITICAL', 'UNKNOWN')
+
+
+def convertToGB(valueStr):
+    value = valueStr[:-1].replace(',', '.')
+    gigs = 0.0
+    if valueStr.endswith('G'):
+        gigs = float(value)
+    elif valueStr.endswith('T'):
+        gigs = float(value) * 1024
+    elif valueStr.endswith('P'):
+        gigs = float(value) * 1024 * 1024
+    elif valueStr.endswith('M'):
+        gigs = float(value) / 1024.0
+    elif valueStr.endswith('K'):
+        gigs = float(value) / (1024.0 * 1024.0)
+    return float(gigs)
+
+
+def raiseState(newState, ret):
```

Comment:
What?

Reply:
This is useful in that you don't have to compare the current chosen state to the state that you want to set.
E.g. if pool health is in critical state, you don't want to override the state to warning if the fragmenation part of the check decides to do so.